src/components/calendarComponent/calendarComponent.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import Calendar from 'react-calendar';
+import 'react-calendar/dist/Calendar.css';
+import './calendar.scss';
+import { Event } from '../../types/components/componentType';
+
+const CalendarComponent: React.FC = () => {
+  const [date, setDate] = useState<Date>(new Date());
+
+  // State for managing events data (with the date as the key)
+  const [events, setEvents] = useState<{ [date: string]: Event }>({});
+
+  // Function to fetch events
+  const fetchEvents = async () => {
+    // Examples
+    const courses: Event[] = [
+      {
+        courseName: 'Introduction to Programming',
+        instructorName: 'John Doe',
+        date: new Date('2024-01-20'),
+      },
+      {
+        courseName: 'Web Development Basics',
+        instructorName: 'Jane Smith',
+        date: new Date('2024-01-15'),
+      },
+      {
+        courseName: 'Web Development Basics',
+        instructorName: 'Jane Smith',
+        date: new Date('2024-01-19'),
+      },
+    ];
+
+    // Reduce courses data to create eventsData object with dates as keys
+    const eventsData = courses.reduce(
+      (acc, course) => {
+        const dateString = course.date.toDateString();
+        acc[dateString] = course;
+        return acc;
+      },
+      {} as { [date: string]: Event },
+    );
+    setEvents(eventsData);
+  };
+
+  // fetch events when the component render
+  useEffect(() => {
+    fetchEvents();
+  }, []);
+
+  // CustomCalendarTile component to render each tile in the calendar
+  const CustomCalendarTile: React.FC<{ date: Date; view: string }> = ({
+    date,
+    view,
+  }) => {
+    const dateString = date.toDateString();
+    const eventData = events[dateString] || {
+      courseName: '',
+      instructorName: '',
+      date,
+    };
+    const { courseName, instructorName } = eventData;
+
+    const truncatedCourseName =
+      courseName.length > 25 ? courseName.substring(0, 25) + '...' : courseName;
+
+    return (
+      <div
+        className={`custom-calendar-tile ${view}`}
+        data-testid="custom-calendar-tile"
+      >
+        <p className="date">{date.getDate()}</p>
+        {truncatedCourseName && (
+          <p className="course-name" data-testid="course-name">
+            {truncatedCourseName}
+          </p>
+        )}
+        {instructorName && (
+          <p className="instructor-name" data-testid="instructor-name">
+            {instructorName}
+          </p>
+        )}
+      </div>
+    );
+  };
```

Comment:
Maybe it will be better make this componet in separete file?

Reply:
Okay, I'll separate it